x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/rules/data/rule_migrations_data_migration_client.ts

```diff
@@ -74,4 +98,49 @@ export class RuleMigrationsDataMigrationClient extends RuleMigrationsDataBaseCli
 
     return [migrationDeleteOperation];
   }
+
+  /**
+   * Updates the last execution parameters for a migration document.
+   */
+  async updateLastExecution({
+    id,
+    lastExecutionParams,
+  }: {
+    id: string;
+    lastExecutionParams: RuleMigrationLastExecution;
+  }): Promise<void> {
+    this.logger.info(`Updating last execution params for migration ${id}`);
+    const index = await this.getIndexName();
+
+    const painlessUpdateScripts: string[] = [];
+
+    for (const [key, value] of Object.entries(lastExecutionParams)) {
+      if (typeof value === 'string') {
+        painlessUpdateScripts.push(`ctx._source.last_execution.${key} = '${value}';`);
```

Comment:
I'm not super sure about the security aspect of `painless` but if we'd treat it like JavaScript, we would want to allow-list `key` here so that it does not allow to set arbitrary props on the object.